src/Nethermind/Nethermind.Core/ConsensusRequests/ConsensusRequest.cs

```diff
@@ -38,46 +39,57 @@ public class ConsensusRequest
 
 public static class ConsensusRequestExtensions
 {
-    public static (int depositCount, int withdrawalRequestCount) GetTypeCounts(this ConsensusRequest[]? requests)
+    public static (int depositCount, int withdrawalRequestCount, int consolidationRequestCount) GetTypeCounts(this ConsensusRequest[]? requests)
     {
         int depositCount = 0;
         int withdrawalRequestCount = 0;
+        int consolidationRequestCount = 0;
         int length = requests?.Length ?? 0;
         for (int i = 0; i < length; i++)
         {
             if (requests![i].Type == ConsensusRequestsType.Deposit)
             {
                 depositCount++;
             }
-            else
+            else if (requests[i].Type == ConsensusRequestsType.WithdrawalRequest)
             {
                 withdrawalRequestCount++;
             }
+            else
+            {
+                consolidationRequestCount++;
+            }
         }
 
-        return (depositCount, withdrawalRequestCount);
+        return (depositCount, withdrawalRequestCount, consolidationRequestCount);
     }
 
-    public static (Deposit[]? deposits, WithdrawalRequest[]? withdrawalRequests) SplitRequests(this ConsensusRequest[]? requests)
+    public static (Deposit[]? deposits, WithdrawalRequest[]? withdrawalRequests, ConsolidationRequest[]? consolidationRequests) SplitRequests(this ConsensusRequest[]? requests)
```

Comment:
out of scope of this PR: I am thinking how can we make this more general and easily extendable?

Reply:
will experiment in different PR